### PR TITLE
tikz: optimize latex render for #1628

### PR DESCRIPTION
**Pull request by liudongmiao**

### CODE REVIEW

File: src/net/sourceforge/plantuml/AbstractPSystem.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ final public long seed() {
 		return getSource().seed();
 	}
 
-	public int getNbImages() {
+	public int getNbImages(FileFormatOption formatOption) {
```
Comment (The-Lum) [inline src/net/sourceforge/plantuml/AbstractPSystem.java:139]: Hi @liudongmiao,

A simple dummy question:
- Why `getNbImages` needs now of `formatOption`?

Regards,
Th.

Comment (arnaudroques) [reply src/net/sourceforge/plantuml/AbstractPSystem.java:139]: I agree with @The-Lum : this is somehow a breaking change, and does not sound very logical to me.

Could you tell us more about this?

Thanks!

Comment (liudongmiao) [reply src/net/sourceforge/plantuml/AbstractPSystem.java]: In AthomMath, the dimension is cached.

https://github.com/plantuml/plantuml/blob/f186e7a6864933629014681345017e6f42ddce8e/src/net/sourceforge/plantuml/klimt/creole/atom/AtomMath.java#L74-L79

There are two way to fix:
- pass the real `StringBounder` when possible (previous commit)
- don't cache for debug (I have reverted the commit, and make a new commit)

**Comment by liudongmiao** [comment]
@arnaudroques I have walked the open issues with latex and examples in https://plantuml.com/zh/ascii-math.
Now the tikz should work now.

The only issue is Embedded Diagram #1926.

**Comment by The-Lum** [comment]
Hello @liudongmiao @arnaudroques 

Here is a global question about package architecture and folder design:
- Is it plan to put together all the file/package (LatexManager, TikzFontDistortion) about latex and tikz in a same folder?

Hint:
- https://github.com/plantuml/plantuml/tree/master/src%2Fnet%2Fsourceforge%2Fplantuml%2Fklimt%2Fdrawing%2Ftikz
Or
- https://github.com/plantuml/plantuml/tree/master/src%2Fnet%2Fsourceforge%2Fplantuml%2Ftikz
[Or in a new folder
- `plantuml/latex` ]

_To debate..._

Regards,
Th.

**Comment by liudongmiao** [comment]
@The-Lum @arnaudroques Yes, it should. And, actually, three things:
- remove `TikzFontDistortion`. `Pragma` to configure `TexSystem` and `TexPreamble` shouldn't be put in file named as `TikzFontDistortion`. Now we use `LaTeX` to get width/height/depth, there should be no font distortion.
- add copyright template in `LatexManager`.
- add test cases. However, these cases would requires runtime TexLive.

**Comment by The-Lum** [comment]
Hello @liudongmiao

FYI, here are some latex test cases from `asciimath2tex`:
- 
https://github.com/plantuml/plantuml/blob/master/test/net/sourceforge/plantuml/math/ASCIIMathTeXImgUnitTest.java

Regards,
Th

**Comment by liudongmiao** [comment]
@The-Lum Not the same. I means the same input latex uml, produces the same tex output.
As the width, height, depth is related to LaTeX, maybe the font and TeX version, so the tex output may not the same.

@The-Lum @arnaudroques Do you know is there any reason to redefind `~`?
https://github.com/plantuml/plantuml/blob/d2b2bcf1722b8705f7f01a556dc96751e7739f7d/src/net/sourceforge/plantuml/tikz/TikzGraphics.java#L398-L401
IMO, it should be converted to `~`, we don't need to ajdust it. If user want to custom it, there is other way.
// Current LatexManager copied this settings too, however, it seems strange.
https://github.com/plantuml/plantuml/blob/bc0becdb83acda69904066e745821032a7690740/src/net/sourceforge/plantuml/LatexManager.java#L115-L116

**Pull request merged**